api/refresh_token_logic.py

```diff
@@ -0,0 +1,83 @@
+from flask import request
+from user_agents import parse
+import requests
+import hashlib
+import time
+import os
+from typing import Optional
+
+class User_info_service:
+    def __init__(self, accass_token_dal, refresh_token_dal):
+        self._access_token_dal = accass_token_dal
+        self._refresh_token_dal = refresh_token_dal
+    def get_user_device(self) -> str:
+        user_agent = request.headers.get("User-Agent", "")
+        parsed_agent = parse(user_agent)
+
+        device_info = {
```

Comment:
Create a model for this and override __str__ method to return your custom string (f"{device_info['device']} | {device_info['os']} | {device_info['browser']}")